src/back-end/lib/db.ts

```diff
@@ -1286,14 +1278,40 @@ export const readManyCWUProposals = tryDb<[Id], CWUProposalSlim[]>(async (connec
       'updatedAt',
       'proponentIndividual',
       'proponentOrganization',
-      'score',
-      'stat.status'
+      'score'
     );
 
   if (!results) {
     throw new Error('unable to read proposals');
   }
 
+  // Read latest status for each proposal
+  for (const proposal of results) {
+    const statusResult = await connection<{ status: CWUProposalStatus }>('cwuProposalStatuses')
+      .where({ proposal: proposal.id })
+      .whereNotNull('status')
+      .orderBy('createdAt', 'desc')
+      .first();
+
+    if (!statusResult) {
+      throw new Error('unable to read proposal status');
+    }
+    proposal.status = statusResult.status;
+  }
+
+  // Read ranks for rankable proposals and apply to existing result set
+  const rankableProposals = results.filter(result => isRankableCWUProposalStatus(result.status));
+  const ranks = await connection
+    .from('cwuProposals')
+    .whereIn('id', rankableProposals.map(r => r.id))
+    .andWhere({ opportunity: id })
+    .select(connection.raw('id, RANK () OVER (ORDER BY score DESC) rank'));
```

Comment:
No score/permissions check because the permissions check is done earlier, I presume?

Reply:
Yeah that's right, the permissions check is done at the readMany resource level